go/pkg/flags/flags.go

```diff
@@ -93,6 +95,14 @@ func NewClientFromFlags(ctx context.Context, opts ...client.Opt) (*client.Client
 		}
 		opts = append(opts, client.RPCTimeouts(timeouts))
 	}
+	var perRPCCreds *client.PerRPCCreds
+	for _, opt := range opts {
+		switch opt.(type) {
```

Comment:
Just an if would suffice?

Reply:
ah actually its not possible to use `.(type)` outside a switch case - https://stackoverflow.com/questions/34820469/can-i-compare-variable-types-with-type-in-golang.